src/main/java/com/commercetools/helper/mapper/impl/PaymentMapperImpl.java

```diff
@@ -0,0 +1,209 @@
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.money.MonetaryAmount;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Stream;
+
+import static com.commercetools.payment.constants.paypalPlus.PaypalPlusPaymentIntent.SALE;
+import static com.commercetools.util.MoneyUtil.getActualShippingCost;
+import static com.commercetools.util.MoneyUtil.getActualTax;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Stream.concat;
+
+@Component
+public class PaymentMapperImpl implements PaymentMapper {
+
+    private final PaypalPlusFormatter paypalPlusFormatter;
+
+    @Autowired
+    public PaymentMapperImpl(@Nonnull PaypalPlusFormatter paypalPlusFormatter) {
+        this.paypalPlusFormatter = paypalPlusFormatter;
+    }
+
+    @Override
+    @Nullable
+    public Payment ctpPaymentToPaypalPlus(@Nullable CtpPaymentWithCart paymentWithCartLike) {
+        if (paymentWithCartLike == null) {
+            return null;
+        }
+
+        final Payment mappedPayment = new Payment();
+
+        mappedPayment.setCart(paymentWithCartLike.getCart().getId());
+        mappedPayment.setIntent(SALE);
+        mappedPayment.setPayer(getPayer(paymentWithCartLike));
+        mappedPayment.setTransactions(getTransactions(paymentWithCartLike));
+        mappedPayment.setRedirectUrls(getRedirectUrls(paymentWithCartLike));
+
+        return mappedPayment;
+    }
+
+    @Nonnull
+    protected Payer getPayer(@Nonnull CtpPaymentWithCart paymentWithCartLike) {
+        return new Payer()
+                .setFundingInstruments(getFundingInstrumentList(paymentWithCartLike))
+                .setPaymentMethod(paymentWithCartLike.getPaymentMethod());
+    }
+
+    @Nonnull
+    protected List<Transaction> getTransactions(@Nonnull CtpPaymentWithCart paymentWithCartLike) {
+        return singletonList(getTransaction(paymentWithCartLike));
+    }
+
+    @Nonnull
+    protected RedirectUrls getRedirectUrls(@Nonnull CtpPaymentWithCart paymentWithCartLike) {
+        return new RedirectUrls()
+                .setReturnUrl(paymentWithCartLike.getReturnUrl())
+                .setCancelUrl(paymentWithCartLike.getCancelUrl());
+    }
+
+    @Nonnull
+    protected List<FundingInstrument> getFundingInstrumentList(@Nonnull CtpPaymentWithCart paymentWithCartLike) {
+        return singletonList(new FundingInstrument()
+                .setCreditCardToken(getCreditCardToken(paymentWithCartLike)));
+    }
+
+    @Nonnull
+    protected CreditCardToken getCreditCardToken(@Nonnull CtpPaymentWithCart paymentWithCartLike) {
+        return new CreditCardToken(paymentWithCartLike.getCreditCardToken());
+    }
+
+    @Nonnull
+    protected Transaction getTransaction(@Nonnull CtpPaymentWithCart paymentWithCartLike) {
+        Transaction transaction = new Transaction();
```

Comment:
I see sometimes you use final, sometimes not. I would suggest this approach: https://stackoverflow.com/a/138990/1293839. WDYT?